packages/@webex/plugin-meetings/src/multistream/remoteMediaManager.ts

```diff
@@ -525,33 +543,96 @@ export class RemoteMediaManager extends EventsScope {
     this.mediaRequestManagers.video.commit();
   }
 
+  /**
+   * Sets which named media group need receiving
+   * @param {MediaType} mediaType of the stream
+   * @param {number} languageCode of the stream
+   * @returns {void}
+   */
+  public async setReceiveNamedMediaGroup(mediaType: MediaType, languageId: number) {
+    if (mediaType !== MediaType.AudioMain) {
+      throw new Error(`cannot set receive named media group which media type is ${mediaType}`);
+    }
+
+    const value = languageId;
+    if (value === this.config.namedMediaGroup?.value) {
+      return;
+    }
+
+    this.config.namedMediaGroup = {
+      type: 1,
+      value,
+    };
+
+    await this.createInterpretationAudioMedia(false);
```

Comment:
I thought we discussed that we don't even need to recreate the RemoteMediaGroup for si audio. If you just add a `setNamedMediaGroup()` method to RemoteMediaGroup and inside it just update the stored named media group and call `this.sendActiveSpeakerMediaRequest()` it will work and client will receive new language on the same stream - see how it's done for example in `setPreferLiveVideo()`.

Reply:
Code changed